src/cpl/nuopc/atm_stream_ndep.F90

```diff
@@ -0,0 +1,254 @@
+         stream_dtlimit      = 1.0e30_r8,                           &
+         stream_tintalgo     = 'linear',                            &
+         stream_name         = 'Nitrogen deposition data ',         &
+         rc                  = rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+       call ESMF_Finalize(endflag=ESMF_END_ABORT)
+    end if
+
+  end subroutine stream_ndep_init
+
+  !================================================================
+  subroutine stream_ndep_check_units( stream_fldFileName_ndep)
+
+    !--------------------------------------------------------
+    ! Check that units are correct on the file and if need any conversion
+    !--------------------------------------------------------
+
+     use cam_pio_utils , only : cam_pio_createfile, cam_pio_openfile, cam_pio_closefile, pio_subsystem
+     use pio           , only : file_desc_t, io_desc_t, var_desc_t, pio_double, pio_def_dim
+     use pio           , only : pio_bcast_error, pio_seterrorhandling, pio_inq_varid, pio_get_att
+     use pio           , only : PIO_NOERR, PIO_NOWRITE
+
+    ! Arguments
+    character(len=*), intent(in)  :: stream_fldFileName_ndep  ! ndep filename
+    !
+    ! Local variables
+    type(file_desc_t) :: File     ! NetCDF filehandle for ndep file
+    type(var_desc_t)  :: vardesc  ! variable descriptor
+    integer           :: ierr     ! error status
+    integer           :: err_handling ! temporary
+    character(len=CS) :: ndepunits! ndep units
+    !-----------------------------------------------------------------------
+
+    call cam_pio_openfile( File, trim(stream_fldFileName_ndep), PIO_NOWRITE)
+    call pio_seterrorhandling(File, PIO_BCAST_ERROR, err_handling)
+    ierr = pio_inq_varid(File, stream_varlist_ndep(1), vardesc)
+    if (ierr /= PIO_NOERR) then
+       call endrun(' ERROR finding variable: '//trim(stream_varlist_ndep(1))//" in file: "// &
+            trim(stream_fldFileName_ndep)//errMsg(sourcefile, __LINE__))
+    else
+       ierr = PIO_get_att(File, vardesc, "units", ndepunits)
+    end if
+    call pio_seterrorhandling(File, err_handling)
+    call cam_pio_closefile(File)
+
+    ! Now check to make sure they are correct
+    if (.not. trim(ndepunits) == "g(N)/m2/s"  )then
+       call endrun(' ERROR in units for nitrogen deposition equal to: '//trim(ndepunits)//" not units expected"// &
+            errMsg(sourcefile, __LINE__))
+    end if
+
+  end subroutine stream_ndep_check_units
+
+  !================================================================
+  subroutine stream_ndep_interp(cam_out, rc)
+
+    use dshr_methods_mod , only : dshr_fldbun_getfldptr
+    use dshr_strdata_mod , only : shr_strdata_advance
+    use camsrfexch       , only : cam_out_t
+    use time_manager     , only : get_curr_date
+    use physics_grid     , only : columns_on_task
+
+    ! input/output variables
+    type(cam_out_t) , intent(inout)  :: cam_out(:)
```

Comment:
should this no longer be an array since we're no longer dealing with chunks?

Reply:
Good catch!  Extra dimension removed.